### PR TITLE
[ISSUE #367] Refactor connector

### PR DESCRIPTION
**Pull request by ruanwenjun**
1.support load plugin from eventMeshPluginDir
2.remove connector plugin from runtime

In local develop, we can add dependency of `eventmesh-connector-rocketmq` in `build.gradle`.
And remove the plugin when packing.

### CODE REVIEW

**Comment by qqeasonchen** [comment]
@ruanwenjun Please hellp to fix these new coming up conflicts.

**Comment by ruanwenjun** [comment]
> @ruanwenjun Please hellp to fix these new coming up conflicts.

Done

File: eventmesh-starter/build.gradle [hunk 1]
```diff
@@ -16,7 +16,7 @@
  */
 
 dependencies {
-    implementation project(":eventmesh-runtime"), project(":eventmesh-connector-rocketmq")
-    testImplementation project(":eventmesh-runtime"), project(":eventmesh-connector-rocketmq")
+    implementation project(":eventmesh-runtime")
```
[the remark this answers is not in the document]
Comment (ruanwenjun) [reply eventmesh-starter/build.gradle:19]: Yes, if the starter module is only used to package dependencies, we can remove this module, but not sure if other things will need to be done in the future before runtime starting.

File: settings.gradle [hunk 2]
```diff
@@ -18,11 +18,12 @@
...
 include 'eventmesh-runtime'
-include 'eventmesh-connector-rocketmq'
 include 'eventmesh-sdk-java'
 include 'eventmesh-common'
 include 'eventmesh-connector-api'
 include 'eventmesh-starter'
 include 'eventmesh-test'
 include 'eventmesh-spi'
+include 'eventmesh-connector-plugin'
+include 'eventmesh-connector-plugin:eventmesh-connector-rocketmq'
```
[the remark this answers is not in the document]
Comment (ruanwenjun) [reply settings.gradle:29]: Yes, we will not package the plugin into runtime, the all plugins will be packaged into jars, and copy to plugin directory.
We can configure the plugin directory.

**Comment by qqeasonchen** [comment]
@ruanwenjun Did you test this already?

**Comment by ruanwenjun** [comment]
> @ruanwenjun Did you test this already?

Yes

**Comment by qqeasonchen** [comment]
@ruanwenjun conflicts needs to be resolved.

**Comment by Jackzeng1224** [comment]
@ruanwenjun  hello,
When I was using eventmesh-connector-plugin, I found that rocketMq could not be loaded. I called rocketMq according to the previous method. Is there any other places that need to be configured?
image

**Comment by ruanwenjun** [comment]
@Jackzeng1224 Which branch do you use? Can you get the log 
```java
initialize extension instance success, extensionType: interface org.apache.eventmesh.api.producer.MeshMQProducer, extensionName: rocketmq
``` 
at `StartUp` log ?

**Comment by ruanwenjun** [comment]
> @ruanwenjun conflicts needs to be resolved.

@qqeasonchen Thanks, done.

**Comment by Jackzeng1224** [comment]
> ```java
> initialize extension instance success,
> ```

@ruanwenjun   hello, The branch I am in is https://github.com/ruanwenjun/incubator-eventmesh/tree/dev_wenjun_refactorPlugin.  I haven't seen it so far   “initialize extension instance success, extensionType: interface org.apache.eventmesh.api.producer.MeshMQProducer, extensionName: rocketmq”

image

**Comment by Jackzeng1224** [comment]
> > ```java
> > initialize extension instance success,
> > ```
> 
> @ruanwenjun hello, The branch I am in is https://github.com/ruanwenjun/incubator-eventmesh/tree/dev_wenjun_refactorPlugin. I haven't seen it so far “initialize extension instance success, extensionType: interface org.apache.eventmesh.api.producer.MeshMQProducer, extensionName: rocketmq”
> 
> image

@ruanwenjun   hello，I think this file configuration is not added.
image

**Comment by ruanwenjun** [comment]
@Jackzeng1224 Yes, now the SPI module will load plugin from `dist/plugin/connector`(this can be changed by environment) and classpath. In local develop, you can execute `dist` task and `copyConnectorPlugin` to install the connector plugin.
Or you can add the `rocketmq` plugin module in `build.gradle`.

**Comment by ruanwenjun** [comment]
@Jackzeng1224 In our design, we don't want to package all plugins in runtime jar. 
The plugin is aim to separate the implementation and provide better scalability, user can choose to load the plugin by config.

**Comment by Jackzeng1224** [comment]
> @Jackzeng1224 Yes, now the SPI module will load plugin from `dist/plugin/connector`(this can be changed by environment) and classpath. In local develop, you can execute `dist` task and `copyConnectorPlugin` to install the connector plugin.
> Or you can add the `rocketmq` plugin module in `build.gradle`.
In the future, add the plug-in to eventmesh's build.gradle.

**Comment by ruanwenjun** [comment]
@Jackzeng1224 How is it going, do you meet other problem?

**Comment by Jackzeng1224** [comment]
> @Jackzeng1224 进展如何，您是否遇到其他问题？

No. Very well designed 👍

**Comment by qqeasonchen** [comment]
When is the doc update? 

> > @ruanwenjun Did you test this already?
> 
> Yes, the documentation of SPI will be added latter.

Have the docs updated now?

**Comment by ruanwenjun** [comment]
@qqeasonchen Yes, the docs has been updated.

File: docs/cn/instructions/eventmesh-runtime-quickstart.zh-CN.md [hunk 3]
```diff
@@ -57,14 +57,20 @@ sh start.sh
 ![project-structure](../../images/project-structure.png)
 
 - eventmesh-common : eventmesh公共类与方法模块
-- eventmesh-connector-api : eventmesh插件接口定义模块
-- eventmesh-connector-rocketmq : eventmesh rocketmq插件模块
+- eventmesh-connector-api : eventmesh connector插件接口定义模块
+- eventmesh-connector-plugin : eventmesh connector插件模块
```
Comment (qqeasonchen) [inline docs/cn/instructions/eventmesh-runtime-quickstart.zh-CN.md:61]: eventmesh-connector-api was moved into eventmesh-connector-plugin，the description and picture are not updated

File: style/checkStyle.xml [hunk 4]
```diff
@@ -44,4 +50,9 @@
       <property name="format" value="^(org)\.apache(\.[a-zA-Z][a-zA-Z0-9]*)+$"/>
     </module>
   </module>
+
+  <module name="BeforeExecutionExclusionFileFilter">
+    <property name="fileNamePattern" value="./eventmesh-runtime/conf/sChat2.jks$"/>
```
Comment (SteveYurongSu) [inline style/checkStyle.xml:55]: Don't forget to remove this module :D

File: docs/cn/instructions/eventmesh-runtime-quickstart.zh-CN.md (hunk 3 above)
Comment (ruanwenjun) [reply docs/cn/instructions/eventmesh-runtime-quickstart.zh-CN.md:61]: Ok, I will submit a pr to fix.

File: style/checkStyle.xml (hunk 4 above)
Comment (ruanwenjun) [reply style/checkStyle.xml:55]: Ok

File: docs/cn/instructions/eventmesh-runtime-quickstart.zh-CN.md (hunk 3 above)
Comment (qqeasonchen) [reply docs/cn/instructions/eventmesh-runtime-quickstart.zh-CN.md:61]: > Ok, I will submit a pr to fix.

See #323 , i want to let the student to fix it :)

Comment (ruanwenjun) [reply docs/cn/instructions/eventmesh-runtime-quickstart.zh-CN.md:61]: Great

**Pull request merged**